core/src/main/java/smile/vq/SOM.java

```diff
@@ -391,6 +391,19 @@ public SOM(double[][] data, int width, int height) {
     }
 
     /**
+     * Returns the cluster labels for each Neuron.
+     */
+    public int[][] clabels() {
+        int[][] clabels = new int[height][width];
```

Comment:
We should check if y is null at the beginning and throw an IllegalStateException if it is. Note y is only created after the method partition is called.